### PR TITLE
Add hyperlinks to diagnostic documentation url

### PR DESCRIPTION
**Pull request by timotheeguerin**
Make the diagnostic code an hyperlink that point to the documentation url

image

### CODE REVIEW

File: packages/compiler/src/core/linter.ts [hunk 1]
```diff
@@ -21,9 +21,6 @@ import {
 export interface Linter {
   extendRuleSet(ruleSet: LinterRuleSet): Promise<readonly Diagnostic[]>;
   lint(): readonly Diagnostic[];
-
-  /** @internal */
-  getRuleUrl(ruleId: string): string | undefined;
```
Comment (timotheeguerin) [inline packages/compiler/src/core/linter.ts:26]: short lived internal api :D turns out needing the program to get the url is very limiting so its easier if we allow each diagnostic to include a url

File: .chronus/changes/feature-cli-hyperlinks-2024-7-12-20-48-20.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+# Change versionKind to one of: internal, fix, dependencies, feature, deprecation, breaking
+changeKind: feature
+packages:
+  - "@typespec/compiler"
+---
+
+Diagnostic logged to the terminal now have clickable hyperlink to the diagnostic documentation url if applicable.
```
Comment (markcowl) [inline .chronus/changes/feature-cli-hyperlinks-2024-7-12-20-48-20.md:8]:
```suggestion
Diagnostics logged to the terminal now have a clickable hyperlink to the diagnostic documentation url if applicable.
```

File: packages/compiler/src/core/linter.ts (hunk 1 above)
Comment (markcowl) [reply packages/compiler/src/core/linter.ts:26]: Since this is exported, shouldn't we deprecate it now and remove later?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/compiler/src/core/linter.ts (hunk 1 above)
Comment (timotheeguerin) [reply packages/compiler/src/core/linter.ts:26]: I just added it this morning, also I don't think anything marked with `@internal` should have to go through this process. Anything marked with `@internal` won't have its type generated

**Review by markcowl — APPROVED** [review]

**Pull request merged**